File: tests/fixtures/ecosystems/dart_comprehensive/basic.dart
/// Basic Dart constructs.
library comprehensive;

String greet(String name) => 'Hello, $name!';

int add(int a, int b) => a + b;

double? divide(double a, double b) {
  if (b == 0) return null;
  return a / b;
}

List<T> transform<T>(List<T> items, T Function(T) fn) {
  return items.map(fn).toList();
}

const String version = '1.0.0';

class Config {
  final String env;
  final bool debug;

  Config({this.env = 'development', this.debug = false});

  bool get isProduction => env == 'production';
}

File: tests/fixtures/ecosystems/php_comprehensive/basic.php
<?php
namespace Comprehensive;

function greet(string $name): string {
    return "Hello, {$name}!";
}

function add(int $a, int $b): int {
    return $a + $b;
}

class Config {
    private string $env;
    private bool $debug;

    public function __construct(string $env = 'development') {
        $this->env = $env;
        $this->debug = $env !== 'production';
    }

    public function getEnv(): string {
        return $this->env;
    }

    public function isDebug(): bool {
        return $this->debug;
    }
}
